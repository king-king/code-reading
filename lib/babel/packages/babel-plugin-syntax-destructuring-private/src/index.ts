import { declare } from "@babel/helper-plugin-utils";

export default declare(api => {
  api.assertVersion(7);

  return {
    name: "syntax-destructuring-private",

    manipulateOptions(_: any, parserOpts: { plugins: string[] }) {
      parserOpts.plugins.push("destructuringPrivate");
    },
  };
});
