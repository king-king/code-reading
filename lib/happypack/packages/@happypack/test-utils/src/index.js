var sinon = require('sinon');
var chai = require('chai');
var composer = require('@happypack/webpack-config-composer');

sinon.assert.expose(chai.assert, { prefix: "" });

exports.VERSION_1 = /^1/;
exports.VERSION_2 = /^2/;
exports.VERSION_3 = /^3/;
exports.HAPPY_LOADER_PATH = require('./root').join('lib/HappyLoader.js');

exports.injectRPCHandler = require('./RPCHandlerShim').set;

exports.assert = require('chai').assert;
exports.fixturePath = require('./fixturePath');
exports.fixture = require('./fixture');
exports.assertNoWebpackErrors = require('./assertNoWebpackErrors');
exports.createIntegrationSuite = require('./createIntegrationSuite');

exports.composeWebpackConfig = function(config) {
  return composer.composeWebpackConfig(composer.getWebpackVersion())(config);
};

exports.getWebpackVersion = composer.getWebpackVersion;
exports.getModuleLoaders = require('./getModuleLoaders');
exports.multiWebpackAssert = require('./multiWebpackAssert');

