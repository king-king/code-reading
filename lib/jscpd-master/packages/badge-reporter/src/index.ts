import {IClone, IOptions, IStatistic} from '@jscpd/core';
import {IReporter} from "@jscpd/finder";
import {badgen} from "badgen";
import {join} from 'path';
import {ensureDirSync, writeFileSync} from "fs-extra";
import {green} from "colors/safe";

export default class BadgeReporter implements IReporter {
  constructor(private options: IOptions) {
  }

  public report(clones: IClone[], statistic: IStatistic): void {
    const badgeOptions = this.options.reportersOptions ? this.options.reportersOptions.badge || {} : {};
    if (this.options.output) {
      const badge = badgen({
        color: this.getColor(statistic),
        status: this.getStatus(statistic),
        subject: 'Copy/Paste',
        ...badgeOptions
      });
      const path = badgeOptions.path ? badgeOptions.path : join(this.options.output, 'jscpd-badge.svg');
      ensureDirSync(this.options.output);
      writeFileSync(path, badge);
      console.log(green(`Badge saved to ${path}`));
    }
  }

  public getStatus(statistic: IStatistic): string {
    return statistic ? statistic.total.percentage + '%' : 'N/A'
  }

  public getColor(statistic: IStatistic): string {
    if (this.options.threshold === undefined) {
      return 'grey';
    }
    return statistic.total.percentage < this.options.threshold ? 'green' : 'red';
  }
}
