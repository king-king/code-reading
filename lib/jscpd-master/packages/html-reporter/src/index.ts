import {join} from 'path';
import {IClone, IOptions, IStatistic} from '@jscpd/core';
import {IReporter, JsonReporter} from "@jscpd/finder";
import {copySync, writeFileSync, readFileSync} from "fs-extra";
import {green, red} from "colors/safe";

export default class HtmlReporter implements IReporter {
  constructor(private options: IOptions) {
  }

  public report(clones: IClone[], statistic: IStatistic): void {
    const jsonReporter = new JsonReporter(this.options);
    const json = jsonReporter.generateJson(clones, statistic);
    if (this.options.output) {
      const src = join(__dirname, '../html/');
      const destination = join(this.options.output, 'html/');
      try {
        copySync(src, destination, {overwrite: true});
        const index = join(destination, 'index.html');
        const html = readFileSync(index).toString();
        writeFileSync(index, html.replace(
          '<body>',
          `<body><script>
                       // <!--
                       window.initialData = ${JSON.stringify(json, null, '  ')};
                       // -->
                       </script>`
        ))
        writeFileSync(join(destination, 'jscpd-report.json'), JSON.stringify(json, null, '  '));
        console.log(green(`HTML report saved to ${join(this.options.output, 'html/')}`));
      } catch (e) {
        console.log(red(e))
      }
    }
  }
}
