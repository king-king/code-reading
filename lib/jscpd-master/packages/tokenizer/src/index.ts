import {IOptions, ITokenizer, ITokensMap} from '@jscpd/core';
import {createTokenMapBasedOnCode} from './tokenize';

export * from './interfaces';
export * from './tokenize';
export * from './token-map';
export * from './formats';

export class Tokenizer implements ITokenizer {
  generateMaps(id: string, data: string, format: string, options: Partial<IOptions>): ITokensMap[] {
    return createTokenMapBasedOnCode(id, data, format, options);
  }
}
