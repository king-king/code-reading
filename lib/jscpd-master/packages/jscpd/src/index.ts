import {getDefaultOptions, IClone, IOptions, IStore, Statistic} from '@jscpd/core';
import {grey, italic} from 'colors/safe';
import {EntryWithContent, getFilesToDetect, InFilesDetector} from '@jscpd/finder';
import {initCli, initOptionsFromCli} from './init';
import {printFiles, printOptions, printSupportedFormat} from './print';
import {createHash} from "crypto";
import {getStore} from './init/store';
import {getSupportedFormats, IMapFrame, Tokenizer} from '@jscpd/tokenizer';
import {registerReporters} from './init/reporters';
import {registerSubscribers} from './init/subscribers';
import {registerHooks} from './init/hooks';

const TIMER_LABEL = 'Detection time:';

export const detectClones = (opts: IOptions, store: IStore<IMapFrame> | undefined = undefined) => {
  const options = {...getDefaultOptions(), ...opts};
  options.format = options.format || getSupportedFormats();

  const files: EntryWithContent[] = getFilesToDetect(options);
  const hashFunction = (value: string): string => {
    return createHash('md5').update(value).digest('hex')
  }
  options.hashFunction = options.hashFunction || hashFunction;
  const currentStore: IStore<IMapFrame> = store || getStore(options.store);
  const statistic = new Statistic(options);
  const tokenizer = new Tokenizer();
  const detector = new InFilesDetector(tokenizer, currentStore, statistic, options);

  registerReporters(options, detector);
  registerSubscribers(options, detector);
  registerHooks(options, detector);

  if (!options.silent) {
    console.time(italic(grey(TIMER_LABEL)));
  }
  return detector.detect(files).then((clones: IClone[]) => {
    if (!options.silent) {
      console.timeEnd(italic(grey(TIMER_LABEL)));
    }
    return clones;
  });
}

export function jscpd(argv: string[]): Promise<IClone[]> {
  const packageJson = require(__dirname + '/../package.json');

  const cli = initCli(packageJson, argv);

  const options: IOptions = initOptionsFromCli(cli);

  if (options.list) {
    printSupportedFormat();
  }

  if (options.debug) {
    printOptions(options);
  }

  if (!options.path || options.path.length === 0) {
    options.path = [process.cwd()];
  }

  if (options.debug) {
    const files: EntryWithContent[] = getFilesToDetect(options);
    printFiles(files);
    return Promise.resolve([]);
  } else {
    const store = getStore(options.store);
    return detectClones(options, store).finally(() => {
      store.close();
    });
  }
}

