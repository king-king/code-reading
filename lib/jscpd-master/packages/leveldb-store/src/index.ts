import {IStore} from '@jscpd/core';
import {IMapFrame} from '@jscpd/tokenizer';
import {ensureDirSync} from 'fs-extra';
import * as rimraf from 'rimraf';

const level = require('level');

export default class LevelDbStore implements IStore<IMapFrame> {
  private name: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  private dbs: Record<string, any> = {};

  get(key: string): Promise<IMapFrame> {
    return this.dbs[this.name].get(key).then((value: string) => JSON.parse(value));
  }

  namespace(name: string): void {
    this.name = name;
    if (!(name in this.dbs)) {
      const path = `.jscpd/${name}`;
      rimraf.sync(path);
      ensureDirSync(path);
      this.dbs[name] = level(path);
    }
  }

  set(key: string, value: IMapFrame): Promise<IMapFrame> {
    return this.dbs[this.name].put(key, JSON.stringify(value));
  }

  close(): void {
    Object.entries(this.dbs).forEach(([name, db]) => {
      db.close(() => {
        rimraf('.jscpd/' + name, {maxBusyTries: 10}, (err) => {
          if (err) {
            console.log(err);
          }
        });
      });
    });
    rimraf.sync('.jscpd');
  }
}
