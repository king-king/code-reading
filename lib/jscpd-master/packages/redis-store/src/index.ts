import {IStore} from '@jscpd/core';
import {IMapFrame} from '@jscpd/tokenizer';

const Redis = require("ioredis");

export default class RedisStore implements IStore<IMapFrame> {
  private name: string;
  private redis

  constructor() {
    this.redis = new Redis();
  }

  close(): void {
    this.redis.disconnect();
  }

  get(key: string): Promise<IMapFrame> {
    return this.redis.get(this.name + ':' + key).then(value => {
      if (!value) {
        throw new Error('not found')
      }
      return JSON.parse(value)
    });
  }

  namespace(name: string): void {
    this.name = name;
  }

  set(key: string, value: IMapFrame): Promise<IMapFrame> {
    return this.redis.set(this.name + ':' + key, JSON.stringify(value));
  }
}
