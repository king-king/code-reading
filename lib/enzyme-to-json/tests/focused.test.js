/* eslint-env jest */

import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from '../src';

Enzyme.configure({ adapter: new Adapter() });

const MyComponent = props => (
    <div className={`my-component ${props.className}`}>
        <h3>Component Title</h3>
        <span>{props.children}</span>
    </div>
);

it('renders the right title', () => {
    const wrapper = shallow(<MyComponent className="strong-class" />);

    expect(shallowToJson(wrapper.find('h3'))).toMatchSnapshot();
});

it('renders a `strong` correctly', () => {
    const wrapper = shallow(
        <MyComponent className="strong-class">
            <strong>Hello World!</strong>
        </MyComponent>,
    );

    expect(shallowToJson(wrapper.find('span').first())).toMatchSnapshot();
});

it('renders a `span` correctly', () => {
    const wrapper = shallow(
        <MyComponent className="span-class">
            <span>Hello World!</span>
        </MyComponent>,
    );

    expect(shallowToJson(wrapper.find('span').first())).toMatchSnapshot();
});
