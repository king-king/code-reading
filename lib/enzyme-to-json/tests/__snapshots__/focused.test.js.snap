// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a \`span\` correctly 1`] = `
<span>
  <span>
    Hello World!
  </span>
</span>
`;

exports[`renders a \`strong\` correctly 1`] = `
<span>
  <strong>
    Hello World!
  </strong>
</span>
`;

exports[`renders the right title 1`] = `
<h3>
  Component Title
</h3>
`;
